pulse-rs/src/stream.rs

```diff
@@ -0,0 +1,413 @@
+    }
+
+    pub fn is_suspended(&self) -> Result<bool> {
+        let r = unsafe { ffi::pa_stream_is_suspended(self.raw_mut()) };
+        error_result!(r != 0, r)
+    }
+
+    pub fn is_corked(&self) -> Result<bool> {
+        let r = unsafe { ffi::pa_stream_is_corked(self.raw_mut()) };
+        error_result!(r != 0, r)
+    }
+
+    pub fn connect_playback<'a, D, A, V, S>(&self,
+                                            dev: D,
+                                            attr: A,
+                                            flags: StreamFlags,
+                                            volume: V,
+                                            sync_stream: S)
+                                            -> Result<()>
+        where D: Into<Option<&'a CStr>>,
+              A: Into<Option<&'a BufferAttr>>,
+              V: Into<Option<&'a CVolume>>,
+              S: Into<Option<&'a mut Stream>>
+    {
+        let r = unsafe {
+            ffi::pa_stream_connect_playback(self.raw_mut(),
+                                            str_to_ptr(dev.into()),
+                                            to_ptr(attr.into()),
+                                            flags.into(),
+                                            to_ptr(volume.into()),
+                                            map_to_mut_ptr(sync_stream.into(), |p| p.0))
+        };
+        error_result!((), r)
+    }
+
+    pub fn connect_record<'a, D, A>(&self, dev: D, attr: A, flags: StreamFlags) -> Result<()>
+        where D: Into<Option<&'a CStr>>,
+              A: Into<Option<&'a BufferAttr>>
+    {
+        let r = unsafe {
+            ffi::pa_stream_connect_record(self.raw_mut(),
+                                          str_to_ptr(dev.into()),
+                                          to_ptr(attr.into()),
+                                          flags.into())
+        };
+        error_result!((), r)
+    }
+
+    pub fn disconnect(&self) -> Result<()> {
+        let r = unsafe { ffi::pa_stream_disconnect(self.raw_mut()) };
+        error_result!((), r)
+    }
+
+    pub fn begin_write(&self, req_bytes: usize) -> Result<(*mut c_void, usize)> {
+        let mut data: *mut c_void = ptr::null_mut();
+        let mut nbytes = req_bytes;
+        let r = unsafe { ffi::pa_stream_begin_write(self.raw_mut(), &mut data, &mut nbytes) };
+        error_result!((data, nbytes), r)
+    }
+
+    pub fn cancel_write(&self) -> Result<()> {
+        let r = unsafe { ffi::pa_stream_cancel_write(self.raw_mut()) };
+        error_result!((), r)
+    }
+
+    pub fn write(&self,
+                 data: *const c_void,
+                 nbytes: usize,
+                 offset: i64,
+                 seek: SeekMode)
+                 -> Result<()> {
+        let r = unsafe {
+            ffi::pa_stream_write(self.raw_mut(), data, nbytes, None, offset, seek.into())
+        };
+        error_result!((), r)
+    }
+
+    pub fn write_with_free<CB>(&self,
```

Comment:
Call it write_with_free_callback? But it doesn't look like it's used, so can we just remove it?